fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/EliminateGroupByConstantRule.java

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.optimizer.rule.transformation;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.starrocks.sql.optimizer.OptExpression;
+import com.starrocks.sql.optimizer.OptimizerContext;
+import com.starrocks.sql.optimizer.operator.OperatorType;
+import com.starrocks.sql.optimizer.operator.logical.LogicalAggregationOperator;
+import com.starrocks.sql.optimizer.operator.logical.LogicalProjectOperator;
+import com.starrocks.sql.optimizer.operator.pattern.Pattern;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.rewrite.ReplaceColumnRefRewriter;
+import com.starrocks.sql.optimizer.rule.RuleType;
+
+import java.util.List;
+import java.util.Map;
+
+public class EliminateGroupByConstantRule extends TransformationRule {
+
+    public static final EliminateGroupByConstantRule INSTANCE = new EliminateGroupByConstantRule();
+
+    private EliminateGroupByConstantRule() {
+        super(RuleType.TF_ELIMINATE_GROUP_BY_CONSTANT,
+                Pattern.create(OperatorType.LOGICAL_AGGR, OperatorType.LOGICAL_PROJECT));
+    }
+
+    @Override
+    public boolean check(OptExpression input, OptimizerContext context) {
+        LogicalAggregationOperator aggOp = input.getOp().cast();
+        LogicalProjectOperator projectOp = input.inputAt(0).getOp().cast();
+        Map<ColumnRefOperator, ScalarOperator> columnRefMap = projectOp.getColumnRefMap();
+        if (aggOp.getGroupingKeys().isEmpty() || aggOp.getAggregations().isEmpty()) {
+            return false;
+        }
+
+        for (ColumnRefOperator groupByCol : aggOp.getGroupingKeys()) {
+            if (!columnRefMap.get(groupByCol).isConstant()) {
```

Comment:
What about partial const group by keys? suck as `group by a, b, 1.0, 'a'`? We can always do this optimization once there are constant operators?

Reply:
How can we distinguish between constant operators and placeholders?
eg:
```
// where 1 reprensents `a`
select a, sum(b) from tbl group by 1;
```

Reply:
> How can we distinguish between constant operators and placeholders? eg:
> 
> ```
> // where 1 reprensents `a`
> select a, sum(b) from tbl group by 1;
> ```

will handle it in analyze phase

Reply:
> What about partial const group by keys? suck as group by a, b, 1.0, 'a'? We can always do this optimization once there are constant operators?
We will handle the constant expr in analyze phase.
This pr mainly focus on group by a column but this column actually is a constant.